sdk/nodejs/provider/experimental/analyzer.ts

```diff
@@ -271,6 +273,46 @@ export class Analyzer {
                 prop.plain = true;
             }
             return prop;
+        } else if (isMapType(type, this.checker)) {
+            const prop: PropertyDefinition = { type: "object" };
+            if (optional) {
+                prop.optional = true;
+            }
+
+            let innerType: typescript.Type;
+            const indexInfo = this.checker.getIndexInfoOfType(type, ts.IndexKind.String);
+            if (type.getSymbol()?.escapedName === "Map") {
+                const typeArguments = (type as typescript.TypeReference).typeArguments;
+                if (typeArguments && typeArguments.length === 2) {
+                    if (!(typeArguments[0].flags & ts.TypeFlags.String)) {
+                        throw new Error(`Expected key of '${this.checker.typeToString(type)}' to be 'string'`);
+                    }
+                    innerType = typeArguments[1];
+                } else {
+                    throw new Error(
+                        `Expected exactly two type arguments in '${this.checker.typeToString(type)}', got '${typeArguments?.length}'`,
+                    );
+                }
+            } else if (type.aliasSymbol?.escapedName === "Record") {
```

Comment:
There is a bit of duplication between this coditionals and isMapType. Any chance to reconcile so that we don't have to keep two pieces in sync?